driver/gamepad.c

```diff
@@ -291,17 +286,8 @@ static int gip_gamepad_op_input(struct gip_client *client, void *data, u32 len)
 
 	if (gip_gamepad_is_elite_controller(gamepad->client))
 	{
-		u8 paddles;
-		struct gip_firmware_version *firmw_ver = (struct gip_firmware_version *)client->firmware_versions->data;
-		if(firmw_ver->major == 5 && firmw_ver->minor >= 11) {
-			//I know this is not ideal...
-			unsigned char *data_bytes = (unsigned char *)data;
-			paddles = data_bytes[64];
-		} else if(firmw_ver->major == 4) {
-			paddles = pkt_xs->paddle_data;
-		}
+		u8 paddles = pkt_xs->paddle_data;
```

Comment:
Surprisingly I could get it working at fw ver 5.1, also verified with usbmon, so I removed the version check.

Reply:
```
ffff8b21f808f240 3476047044 C Ii:3:012:2 0:4 51 = 2000292f 00000000 00007301 97f95efc 9efa0400 00000000 00000000 00000000
ffff8b21f808f240 3476047061 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476050983 C Ii:3:012:2 0:4 21 = 0c002911 00000000 00007301 97f95efc 9efa0400 0a
ffff8b21f808f240 3476050989 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476206947 C Ii:3:012:2 0:4 51 = 20002a2f 00000000 00007301 97f95efc 9efa0000 00000000 00000000 00000000
ffff8b21f808f240 3476206956 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476210904 C Ii:3:012:2 0:4 21 = 0c002a11 00000000 00007301 97f95efc 9efa0000 0a
ffff8b21f808f240 3476210909 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476390945 C Ii:3:012:2 0:4 51 = 20002b2f 00000000 00007301 97f95efc 9efa0100 00000000 00000000 00000000
ffff8b21f808f240 3476390956 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476394904 C Ii:3:012:2 0:4 21 = 0c002b11 00000000 00007301 97f95efc 9efa0100 0a
ffff8b21f808f240 3476394907 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476495017 C Ii:3:012:2 0:4 51 = 20002c2f 00000000 00007301 97f95efc 9efa0000 00000000 00000000 00000000
ffff8b21f808f240 3476495032 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476498924 C Ii:3:012:2 0:4 21 = 0c002c11 00000000 00007301 97f95efc 9efa0000 0a
ffff8b21f808f240 3476498933 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476655058 C Ii:3:012:2 0:4 51 = 20002d2f 00000000 00007301 97f95efc 9efa0800 00000000 00000000 00000000
ffff8b21f808f240 3476655077 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476658915 C Ii:3:012:2 0:4 21 = 0c002d11 00000000 00007301 97f95efc 9efa0800 0a
ffff8b21f808f240 3476658920 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476815011 C Ii:3:012:2 0:4 51 = 20002e2f 00000000 00007301 97f95efc 9efa0000 00000000 00000000 00000000
ffff8b21f808f240 3476815028 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476818989 C Ii:3:012:2 0:4 21 = 0c002e11 00000000 00007301 97f95efc 9efa0000 0a
ffff8b21f808f240 3476818995 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476927018 C Ii:3:012:2 0:4 51 = 20002f2f 00000000 00007301 97f95efc 9efa0200 00000000 00000000 00000000
ffff8b21f808f240 3476927036 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3476930932 C Ii:3:012:2 0:4 21 = 0c002f11 00000000 00007301 97f95efc 9efa0200 0a
ffff8b21f808f240 3476930941 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3477055050 C Ii:3:012:2 0:4 51 = 2000302f 00000000 00007301 97f95efc 9efa0000 00000000 00000000 00000000
ffff8b21f808f240 3477055085 S Ii:3:012:2 -115:4 64 <
ffff8b21f808f240 3477058955 C Ii:3:012:2 0:4 21 = 0c003011 00000000 00007301 97f95efc 9efa0000 0a
```
![Screenshot from 2023-09-27 02-13-16](https://github.com/RenV123/xone/assets/8340896/86ff8034-0699-48b3-beb3-ed3a27d905e6)
<img width="394" alt="image" src="https://github.com/RenV123/xone/assets/8340896/4fc7e5e6-a019-4ef8-82ae-2b80b9dc53c4">